airbyte-ci/connectors/base_images/base_images/python_bases.py

```diff
@@ -0,0 +1,149 @@
+            sha="40582fe697811beb7bfceef2087416336faa990fd7e24984a7c18a86d3423d58",
+            platform=dagger.Platform("linux/amd64"),
+        ),
+        dagger.Platform("linux/arm64"): common.PlatformAwareDockerImage(
+            image_name="python",
+            tag="3.9.18-bookworm",
+            sha="0d132e30eb9325d53c790738e5478e9abffc98b69115e7de429d7c6fc52dddac",
+            platform=dagger.Platform("linux/arm64"),
+        ),
+    }
+
+
+class AirbytePythonConnectorBaseImage(common.AirbyteConnectorBaseImage, ABC):
+    """An abstract class that represents an Airbyte Python base image."""
+
+    image_name: Final[str] = "airbyte-python-connector-base"
+
+    # These env vars were set on all our certified python connectors that were not using this base image
+    EXPECTED_ENV_VARS: Set[str] = {
+        "PYTHON_VERSION",
+        "PYTHON_PIP_VERSION",
+        "PYTHON_GET_PIP_SHA256",
+        "PYTHON_GET_PIP_URL",
+        "AIRBYTE_BASE_BASE_IMAGE",
+        "AIRBYTE_BASE_IMAGE",
+        "HOME",
+        "PATH",
+        "LANG",
+        "GPG_KEY",
+        "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL",
+        "PYTHON_SETUPTOOLS_VERSION",
+        "OTEL_TRACES_EXPORTER",
+        "OTEL_TRACE_PARENT",
+        "TRACEPARENT",
+    }
+
+    async def run_sanity_checks(self):
+        await super().run_sanity_checks()
+        await self.check_env_vars()
+
+    async def check_env_vars(self):
+        try:
+            printenv_output: str = await self.container.with_exec(["printenv"], skip_entrypoint=True).stdout()
+        except dagger.ExecError as e:
+            raise common.SanityCheckError("failed to run printenv.") from e
+        env_vars = set([line.split("=")[0] for line in printenv_output.splitlines()])
+        missing_env_vars = self.EXPECTED_ENV_VARS - env_vars
+        if missing_env_vars:
+            raise common.SanityCheckError(f"missing environment variables: {missing_env_vars}")
+
+
+class _0_1_0(AirbytePythonConnectorBaseImage):
+
+    base_base_image: Final[PythonBase] = PythonBase.PYTHON_3_9
+
+    TIMEZONE: Final[str] = "Etc/UTC"
+    # This should be a final class attribute if the base_base_image attribute is Final
+    EXPECTED_PYTHON_VERSION: Final[str] = "3.9.18"
+    EXPECTED_PIP_VERSION: str = "23.2.1"
+
+    changelog: str = (
+        "Declare our first base image version. It uses Python 3.9.18 on a Debian 11 (Bookworm) system with Pip 23.2.1 and UTC timezone."
+    )
+
+    @property
+    def container(self) -> dagger.Container:
+        return self.base_container.with_exec(["ln", "-snf", f"/usr/share/zoneinfo/{self.TIMEZONE}", "/etc/localtime"]).with_exec(
+            ["pip", "install", "--upgrade", f"pip=={self.EXPECTED_PIP_VERSION}"]
+        )
+
+    async def run_sanity_checks(self):
+        await super().run_sanity_checks()
+        await self.check_python_version()
+        await self.check_pip_version()
+        await self.check_time_zone()
+        await self.check_bash_is_installed()
```

Comment:
TODO: add docstring explaining this checks are from the results of our [analysis](https://github.com/airbytehq/airbyte/blob/9ff3554ec31aedf49e491bb50e880dd76df2e61e/airbyte-ci/connectors/pipelines/Python%20connector%20images%20analysis.ipynb)